prediction_market_agent/agents/microchain_agent/nft_treasury_game/deploy_nft_treasury_game.py

```diff
@@ -71,6 +92,45 @@ def load(self) -> None:
 
         super().load()
 
+    def before_iteration_callback(self) -> None:
+        if (
+            self.agent.history
+            and GameRoundEnd.GAME_ROUND_END_OUTPUT in str(self.agent.history[-1])
+            and get_nft_game_status() == NFTGameStatus.finished
+        ):
```

Comment:
I expected NFTGameStatus to be shared across all agents (e.g. a DB table) - like this works, but curious about your rationale behind it?

Reply:
What other is there to be shared? Once the treasury balance becomes 0, that's shared across all agents from the blockchain itself.

What would be the benefit of saving it in the table? The disadvantage is more complexity and ping (it would take some time after the balance becomes 0, for some cronjob, to check it and update Postgres with it).